exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanMarshaler.java

```diff
@@ -57,6 +58,8 @@ static SpanMarshaler create(SpanData spanData) {
             ? EMPTY_BYTES
             : encodeTraceState(traceState).getBytes(StandardCharsets.UTF_8);
 
+    int flags = spanData.getSpanContext().getTraceFlags().asByte();
```

Comment:
I think we need to convert this to an unsigned int. See equivalent in [LogMarshaler](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/LogMarshaler.java#L128).

Reply:
eek, good call. Stupid signed bytes.

Reply:
Thanks for pointing this out, I think we need this in the disk buffering in contrib as well, so I'll go chase that.